journalpump/senders/websocket.py

```diff
@@ -176,10 +179,27 @@ async def websocket_connect_coro(self):
                 socks_url_parsed.hostname,
                 socks_url_parsed.port,
             )
+        else:
+            # Resolve hostname and pick one address at random
+            # Websockets.connect() and underlying asyncio.loop.create_connection() have some overlapping timeouts
+            # that lead to little bit difficulties with working through all addresses returned by getaddrinfo.
+            # We pick one at random here, and rely our own outer loops to handle retries.
+            present_addrs = [
+                gai_result[4][0]
+                for gai_result in await self.websocket_loop.getaddrinfo(
+                    url_parsed.hostname, 0, type=socket.SOCK_STREAM, proto=socket.IPPROTO_TCP
+                )
```

Comment:
I feel the [4][0] look a bit "magic" and/or obscure even though I suppose these are "well known":

Could the readability of this be improved with destructuring assignment, such as
 
```suggestion
                sockaddr[0] # Get IP address only
                for _, _, _, _, sockaddr in await self.websocket_loop.getaddrinfo(
                    url_parsed.hostname, 0, type=socket.SOCK_STREAM, proto=socket.IPPROTO_TCP
                )
```

or even:
```suggestion
                ip_address
                for _, _, _, _, (ip_address, *_) in await self.websocket_loop.getaddrinfo(
                    url_parsed.hostname, 0, type=socket.SOCK_STREAM, proto=socket.IPPROTO_TCP
                )
```

Reply:
The improvement works fine as long as they don't decide to add a 6th element to the tuple. Maybe an extra *_ would avoid that risk.

Reply:
I adjusted the code with a comment and the first proposed form. getaddrinfo() is used quite widely, changes to the return types would be quite disruptive so I think it's good with the parsing of the 5-tuple.